### DISCUSSION TITLE
Opening a PDF

### DISCUSSION THREAD

**Discussion by joeprivettsmart** [remix-run/remix#4463] [Q&A]
I am trying simply to create a link to a PDF, where the PDF is stored in the application folder structure somewhere. This stackoverflow post summarises my issue:
https://stackoverflow.com/questions/74264985/linking-to-static-asset-pdf-in-remix

**Comment by machour** [comment]
Putting the file in `/public/` should have worked. 

If you create `/public/file.pdf` then you should be able to do a `<a href="/file.pdf">download</a>` to link to it.

**Reply by joeprivettsmart** [reply]
Adding it to the public folder actually does work, just not quite how I want it to. Our application has a root path of `/employee`. so when I add the pdf to the `public` folder, the url that is opened is https://myurl/some-pdf.pdf, whereas I need the path to be https://myurl/employee/some-pdf.pdf

**Reply by machour** [reply]
In this case, you have two choices:

- Configure your http server (nginx, express) so that `myurl/employee` reads from the root path `/employee`, and totally bypass remix

- Create a Remix [resource route](https://remix.run/docs/en/v1/guides/resource-routes) in `app/routes/employee/$path[.pdf].tsx`, read the `$path` from the loader and return a Response with the corresponding file in `/employee`:

```jsx
export async function loader({ params }) {
  const pdf = await getFileContents(params.path);
  return new Response(pdf, {
    status: 200,
    headers: {
      "Content-Type": "application/pdf",
    },
  });
}
```

(Using a resource route would allow you to check for authentication)

**Reply by joeprivettsmart** [reply]
Thanks for the suggestion. a resource route was an option in my mind, but it felt like it was getting unnecessarily complicated for such a simple thing. I'm currently trying to understand how to read the PDF (i.e. where you have a function getFileContents) - any ideas?

**Reply by joeprivettsmart** [reply]
I think this should work:

```import { readFileSync } from 'fs'

const getFileContents = (filename: string) => {
  return readFileSync(`./public/shared-assets/${filename}.pdf`)
}
```

**Reply by machour** [reply]
you can use the node fs module to read the file:

```jsx
const pdf = fs.readFileSync(`/root/employee/${params.path}.pdf`;
```

Make sure to validate `params.path` content before doing that, for security considerations
